rfc-126-custom-configuration-for-dependabot.md

```diff
@@ -0,0 +1,246 @@
+Top 5 quickest libraries to merge: docker-compose (0:12:51), vcr (0:10:44), miller-columns-element (0:05:36), faraday_middleware (0:03:30), nltk (0:01:14)
+```
+</details>
+
+## Proposal
+
+We will configure Dependabot to only raise PRs against libraries that fit within one of three categories. Any updates to libraries which don't fit into one of these categories will be ignored and PRs won't be raised for them.
+
+The three categories are:
+
+- [Security updates]
+- Internal libraries - these are libraries written by us (for example `gds-api-adapters`, `govuk_publishing_components`)
+- Framework libraries - these are libraries which our apps heavily rely on to work (for example `rails`, `rspec-rails`)
+
+These categories have been chosen to ensure we stay up to date on libraries which are important to us while also reducing the number of unnecessary version bumps.
+
+[Security updates]: https://help.github.com/en/github/managing-security-vulnerabilities/configuring-github-dependabot-security-updates#about-github-dependabot-security-updates
+
+### Custom configuration
+
+Dependabot [supports configuration files][dependabot-config] stored in each repo to customise its behaviour. This RFC proposes using a configuration file like the one detailed below in each of our repos.
+
+We've already been trialing this out in [Content Publisher] and [govuk_publishing_components], based on some initial experiements in [Signon].
+
+[dependabot-config]: https://dependabot.com/docs/config-file/
+[Content Publisher]: https://github.com/alphagov/content-publisher/pull/2062
+[govuk_publishing_components]: https://github.com/alphagov/govuk_publishing_components/pull/1564
+[Signon]: https://github.com/alphagov/signon/pull/1426
+
+```yaml
+version: 1
+update_configs:
+  - package_manager: ruby:bundler
+    directory: /
+    update_schedule: daily
+    allowed_updates:
+      # Security updates
+      - match: { update_type: security }
+      - match: { dependency_name: brakeman }
+      # Internal gems
+      - match: { dependency_name: gds-api-adapters }
+      - match: { dependency_name: gds-sso }
+      - match: { dependency_name: govspeak }
+      - match: { dependency_name: govuk_app_config }
+      - match: { dependency_name: govuk_publishing_components }
+      - match: { dependency_name: govuk_schemas }
+      - match: { dependency_name: govuk_sidekiq }
+      - match: { dependency_name: govuk_test }
+      - match: { dependency_name: rubocop-govuk }
+      - match: { dependency_name: plek }
+      - match: { dependency_name: scss_lint-govuk }
+      # Framework gems
+      - match: { dependency_name: factory_bot_rails }
+      - match: { dependency_name: jasmine }
+      - match: { dependency_name: rails }
+      - match: { dependency_name: rspec-rails }
+      - match: { dependency_name: sass-rails }
```

Comment:
We could make use of wildcards and the `dependency_type` field.

```
      - match:
          dependency_name: "govuk*"
          dependency_type: "production"
```

This will include all gems beginning with "govuk" and that aren't in the "development" group.

Not bumping gems used only for local development might cut down on the number of unnecessary bumps.

Reply:
+1 for an asterisk - that's some nice future proofing in there.

I think we should bump dev dependencies:

- We want to keep up-to-date with framework gems like RSpec, which we write a lot of code with.
- We want to keep up-to-date with internal dev gems, like RuboCop GOV.UK. No one wants to raise all those PRs manually.

Reply:
Oh yeah, that's true. Though I wonder if they should be in the `test` group; I'm not sure if `production` includes this (there's no `test` dependabot type).